juju/juju
# Rewritten spaces discovery code

## Description of change
1. Add a method of manually refreshing spaces from substrate (other than restart of jujud) - juju reload-spaces
2. Make sure that at when add-model returns spaces are loaded

## QA steps
1. Add space in MAAS, verify that after juju reload-spaces the space is visible in juju
2. Add subnet in GCE (substrate not supporting spaces), verify that after juju reload-spaces the subnet is visible in juju
3. Make sure that juju add-model foobar; juju -m foobar list-spaces shows the proper spaces (currently it doesn't as there's a timing issue and add-model returns before spaces are loaded)
4. Make sure that after bootstrap both controller and default model have spaces loaded

## Documentation changes
1. New CLI command juju reload-spaces
2. When juju add-model returns spaces are loaded

## Bug reference
https://bugs.launchpad.net/juju/+bug/1662264
https://bugs.launchpad.net/juju/+bug/1623324
